src/main/scala/org/clulab/wm/eidos/groundings/grounders/SRLCompositionalGrounder.scala

```diff
@@ -0,0 +1,188 @@
+    // or else ground them.
+    else {
+      val attachmentTriggers = findAttachmentTriggers(mention.odinMention)
+      groundSentenceSpan(mention.odinMention.sentenceObj, mention.odinMention.start, mention.odinMention.end, attachmentTriggers)
+    }
+  }
+
+
+
+  def groundPredicates(sentence: Sentence, tokenInterval: Interval, exclude: Seq[Int]): Seq[PredicatePackage] = {
+    val srls = sentence.semanticRoles.get
+    val predicates = srls.roots.toSeq
+      // keep only predicates that are within the mention
+      .filter(tokenInterval contains _)
+      // remove the predicates which correspond to our increase/decrease/quantifiers
+      .filterNot(exclude contains _)
+      // start with those closest to the syntactic root of the sentence to begin with "higher level" predicates
+      .sortBy(minGraphDistanceToSyntacticRoot(_, sentence.dependencies.get))
+
+    packagePredicates(predicates, Seq(), Set(), srls, predicates.toSet)
+  }
+
+
+
+  @tailrec
+  private def packagePredicates(remaining: Seq[Int], results: Seq[PredicatePackage], seen: Set[Int], srls: DirectedGraph[String], predicates: Set[Int]): Seq[PredicatePackage] = {
+    remaining match {
+      case Seq() => results
+      case Seq(curr) => results ++ Seq(mkPredicatePackage(curr, srls, predicates))
+      case curr :: rest =>
+        if (seen contains curr) {
+          // we've been here before
+          packagePredicates(rest, results, seen, srls, predicates)
+        } else if (srls.roots contains curr) {
+          // Otherwise, it's new and it's a predicate
+          val packaged = mkPredicatePackage(curr, srls, predicates)
+          packagePredicates(rest, results ++ Seq(packaged), seen ++ Set(curr), srls, predicates)
+        }
+        else {
+          ???
+          // I don't think this should happen
+          // It's not a predicate, meaning it has already been included in some other predicate's package
+          // packagePredicate(rest, results, seen ++ Set(curr), srls, predicates)
+        }
+    }
+  }
+
+  // Ground the predicate and also ground each of the relevant arguments
+  // Return a PredicatePackage object
+  private def mkPredicatePackage(predicate: Int, srls: DirectedGraph[String], validPredicates: Set[Int]): PredicatePackage = {
+    // Otherwise, it's new.  If it's a predicate:
+    val groundedPredicate = groundToken(predicate, validPredicates)
+    // get the arguments
+    val agents = groundArguments(predicate, AGENT_ROLE, srls, validPredicates)
+    val themes = groundArguments(predicate, THEME_ROLE, srls, validPredicates)
+    val others = groundArguments(predicate, OTHER_ROLE, srls, validPredicates)
+    PredicatePackage(groundedPredicate, agents, themes, others)
+  }
+
+  // Find all arguments from a given predicate of a certain role (e.g., A1, A0, etc) and ground them
+  private def groundArguments(predicate: Int, role: String, srls: DirectedGraph[String], validPredicates: Set[Int]): Seq[GroundedToken] = {
```

Comment:
In groundArguments, I think we need to syntactically expand the arguments rather than grounding a single token. This is a limitation of the CoNLL SRL annotation, that is, the argument dependency points to the *syntactic head* of the argument, rather than the whole span. We need to identify the span ourselves, using your expand magic.

Reply:
@BeckySharp: this was super fast, and is super cool. You are awesome.

Reply:
thanks! i had fun!  Your idea to expand is a good one, and i def wasn't thinking about it.  I may do some lightweight expansion (to the chunk) instead of full expansion which may be likely to suck up the other components....
ok, more soon